### PR TITLE
KFSPTS-5689: eliminate spring dependencies for CuCreditMemoServiceImplTest

### PR DESCRIPTION
**Pull request by bhutchinson**

### CODE REVIEW

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java [hunk 1]
```diff
@@ -1,97 +1,250 @@
...
+        EasyMock.replay(accountsPayableService);
+
+        dataDictionaryService = EasyMock.createNiceMock(DataDictionaryService.class);
+        EasyMock.expect(dataDictionaryService.getAttributeMaxLength(DocumentHeader.class, KRADPropertyConstants.DOCUMENT_DESCRIPTION)).andReturn(200);
+        EasyMock.replay(dataDictionaryService);
+
+        dateTimeService = new DateTimeServiceImpl();
+        documentService = new MockDocumentServiceImpl();
+        noteService = EasyMock.createMock(NoteService.class);
+        purapService = EasyMock.createMock(PurapService.class);
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: The mocked noteService and purapService are not being replayed.

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java [hunk 2]
```diff
@@ -1,97 +1,250 @@
...
+        vendorService = new MockVendorServiceImpl();
+
+        creditMemoServiceImpl = PowerMock.createPartialMock(CuCreditMemoServiceImplTest.TestCuCreditMemoServiceImpl.class, "reIndexDocument", "getCreditMemoDocumentById");
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: The mocked creditMemoServiceImpl is only being replayed by some of the unit test methods.

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java [hunk 3]
```diff
@@ -1,97 +1,250 @@
...
 
+    private CuVendorCreditMemoDocument setupVendorCreditMemoDocument() {
+        ArrayList<String> methodNames = new ArrayList<>();
+        for (Method method : VendorCreditMemoDocument.class.getMethods()) {
+            if (!Modifier.isFinal(method.getModifiers()) && !method.getName().startsWith("set") && !method.getName().startsWith("get") && !method.getName().startsWith("is")) {
+                methodNames.add(method.getName());
+            }
+        }
+        IMockBuilder<CuVendorCreditMemoDocument> builder = EasyMock.createMockBuilder(CuVendorCreditMemoDocument.class).addMockedMethods(methodNames.toArray(new String[0]));
+        creditMemoDocument = builder.createNiceMock();
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: The mocked creditMemoDocument is not being replayed.

File: src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java [hunk 4]
```diff
@@ -1,97 +1,250 @@
...
+	    @Override
+        public Note createNoteFromDocument(Document document, String text) {
+            ArrayList<String> methodNames = new ArrayList<>();
+            for (Method method : Note.class.getMethods()) {
+                if (!Modifier.isFinal(method.getModifiers()) && !method.getName().startsWith("set") && !method.getName().startsWith("get")) {
+                    methodNames.add(method.getName());
+                }
+            }
+            IMockBuilder<Note> builder = EasyMock.createMockBuilder(Note.class).addMockedMethods(methodNames.toArray(new String[0]));
+            Note note = builder.createNiceMock();
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/module/purap/document/service/impl/CuCreditMemoServiceImplTest.java]: The mocked note is not being replayed.

**Comment by bhutchinson** [comment]
PowerMock extends other mock libraries like EasyMock: https://github.com/powermock/powermock
Yes, I needed it to mock a private method. As far as the EasyMock.replay() calls - if they aren't needed, why should we add them?

**Comment by bhutchinson** [comment]
After further review, I really just don't want the methods to be called in those services; calling replay requires me to setup the expected method calls. I can do that, but this is easier, and I'm not sure how much value that adds.

**Comment by bhutchinson** [comment]
Thanks Chad, yes I believe that is a good topic for further discussion at one of our team meetings.

**Pull request merged**